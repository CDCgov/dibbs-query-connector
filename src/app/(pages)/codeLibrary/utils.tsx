/**
 * Helper method to extract the source system display name from the url string.
 * Similar to DBServive.stripProtocolAndTLDFromSystemUrl, but builds in specific
 * checks for known string outliers
 * @param system the url string to format
 * @returns the display name of the system, stripped of url prefixes
 */
export const formatSystem = (system: string) => {
  const match = system.match(/https?:\/\/([^\.]+)/);
  let result = match?.[1];

  switch (result) {
    case "hl7":
      const hl7Arr = system.split("/");
      result = hl7Arr[hl7Arr.length - 1];
      if (result == "icd-10-cm") {
        result = result.slice(0, result.indexOf("-cm"));
      }
      return result.toLocaleUpperCase();

    case "www":
      const rxArr = system.split("/");
      return rxArr[rxArr?.length - 1].toLocaleUpperCase();

    default:
      return result?.toUpperCase();
  }
};
