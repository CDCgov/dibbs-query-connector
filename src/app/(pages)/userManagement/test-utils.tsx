import React, { createRef, RefObject } from "react";
import { Tab } from "@/app/ui/designSystem/TabGroup/tabGroup";
import { UserRole, User, UserGroup } from "@/app/models/entities/users";
import UserPermissionsTable from "./components/userPermissionsTable/userPermissionsTable";
import UserGroupsTable from "./components/userGroupsTable/UserGroupsTable";
import { CustomUserQuery } from "@/app/models/entities/query";

// User Mocks
export const mockSuperAdmin = {
  id: "123",
  username: "harrypotter",
  firstName: "Harry",
  lastName: "Potter",
  qcRole: UserRole.SUPER_ADMIN,
};

export const mockAdmin: User = {
  id: "456",
  username: "lilypotter",
  firstName: "Lily",
  lastName: "Potter",
  qcRole: UserRole.ADMIN,
  userGroupMemberships: [
    {
      membershipId: "456_876",
      usergroupName: "Order of the Phoenix",
      usergroupId: "876",
      isMember: true,
    },
  ],
};

export const mockStandard: User = {
  id: "789",
  username: "hermionegranger",
  firstName: "Hermione",
  lastName: "Granger",
  qcRole: UserRole.ADMIN,
  userGroupMemberships: [
    {
      membershipId: "789_012",
      usergroupName: "S.P.E.W.",
      usergroupId: "012",
      isMember: true,
    },
  ],
};

// UserGroup Mocks
export const mockGroupBasic: UserGroup = {
  id: "123",
  name: "Hogwarts",
  memberSize: 1,
  querySize: 0,
};

export const mockGroupMany: UserGroup = {
  id: "456",
  name: "Order of the Phoenix",
  memberSize: 3,
  querySize: 3,
  members: [{} as User, {} as User, {} as User],
  queries: [
    {} as CustomUserQuery,
    {} as CustomUserQuery,
    {} as CustomUserQuery,
  ],
};

const mockGroupSingle = {
  id: "789",
  name: "S.P.E.W.",
  memberSize: 1,
  querySize: 1,
  members: [mockAdmin],
  queries: [{} as CustomUserQuery],
};

export const allGroups = [mockGroupMany, mockGroupSingle];

const mockRefs = createRef<RefObject<HTMLTableRowElement | null>[]>();
mockRefs.current = [];

// TabGroup Mocks
export const mockPermissionsTab: Tab = {
  label: "Users",
  access: ["Super Admin"],
  onClick: jest.fn(),
  renderContent: jest
    .fn()
    .mockReturnValue(
      <UserPermissionsTable
        setUsers={jest.fn()}
        users={[mockAdmin, mockSuperAdmin]}
        fetchGroupMembers={jest.fn()}
        openModal={jest.fn()}
        rowFocusRefs={mockRefs}
        modalData={""}
      />,
    ),
};
export const mockGroupsTab: Tab = {
  label: "User Groups",
  access: ["Super Admin", "Admin"],
  onClick: jest.fn(),
  renderContent: jest
    .fn()
    .mockReturnValue(
      <UserGroupsTable
        userGroups={[mockGroupBasic]}
        fetchGroupMembers={jest.fn().mockReturnValue(mockAdmin)}
        fetchGroupQueries={jest.fn()}
        openModal={jest.fn()}
        modalData={""}
      />,
    ),
};

// Query Mocks
export const mockQueryNoGroups: CustomUserQuery = {
  queryId: "q1",
  queryName: "Test Query 1",
  conditionsList: [],
  valuesets: [],
};

export const mockQueryWithGroups: CustomUserQuery = {
  queryId: "q2",
  queryName: "Test Query 2",
  conditionsList: [],
  valuesets: [],
  groupAssignments: [
    {
      membershipId: `q2_${mockGroupSingle.id}`,
      usergroupName: mockGroupSingle.name,
      usergroupId: mockGroupSingle.id,
      isMember: true,
    },
  ],
};

mockGroupSingle.queries.push(mockQueryWithGroups);

export const mockGroupWithSingleQuery = mockGroupSingle;
