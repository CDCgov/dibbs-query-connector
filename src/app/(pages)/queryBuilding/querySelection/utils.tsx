import { ModalRef } from "@/app/ui/designSystem/modal/Modal";
import { DeleteModal } from "@/app/ui/designSystem/modal/deleteModal";
import { JSX, RefObject } from "react";
import { showToastConfirmation } from "@/app/ui/designSystem/toast/Toast";
import { DataContextValue } from "@/app/utils/DataProvider";
import { deleteQueryById } from "@/app/backend/query-building/service";
import { CustomUserQuery } from "@/app/models/entities/query";

/**
 * Handles deleting a user query.
 * @param queryName - The name of the query to delete.
 * @param queryId - The ID of the query to delete.
 * @param queries - The current list of user queries.
 * @param setQueries - Function to update the state of queries.
 * @param context - The data context used to update shared state.
 */
export const handleDelete = async (
  queryName: string | undefined,
  queryId: string | undefined,
  queries: CustomUserQuery[],
  setQueries: React.Dispatch<
    React.SetStateAction<CustomUserQuery[] | undefined>
  >,
  context: DataContextValue | undefined,
) => {
  if (queryId) {
    const result = await deleteQueryById(queryId);
    if (result.success) {
      showToastConfirmation({
        body: `${queryName} has been deleted.`,
        variant: "success",
        duration: 2000,
      });
      const updatedQueries = queries.filter(
        (query) => query.queryId !== queryId,
      );
      setQueries(updatedQueries);

      if (context?.setData) {
        context.setData(updatedQueries);
      }
    } else {
      showToastConfirmation({
        heading: `Something went wrong`,
        body: `${queryName} couldn't be deleted. Please try again or contact us if the error persists`,
        variant: "error",
        duration: 2000,
      });
    }
  } else {
    showToastConfirmation({
      heading: `Something went wrong`,
      body: `${queryName} couldn't be deleted. Please try again or contact us if the error persists`,
      variant: "error",
      duration: 2000,
    });
  }
};

/**
 * Confirms the deletion of a user query by toggling a modal.
 * @param queryName - The name of the query to confirm deletion for.
 * @param queryId - The ID of the query to confirm deletion for.
 * @param setSelectedQuery - Function to set the currently selected query for deletion.
 * @param modalRef - Reference to the modal component.
 */
export const confirmDelete = (
  queryName: string,
  queryId: string,
  setSelectedQuery: React.Dispatch<React.SetStateAction<SelectedQueryDetails>>,
  modalRef: RefObject<ModalRef | null>,
) => {
  setSelectedQuery({ queryName, queryId });
  modalRef.current?.toggleModal();
};

/**
 * Copies the query ID to the clipboard.
 * @param queryName - The name of the query to copy the ID for.
 * @param queryId - The ID of the query to copy.
 */
export const handleCopy = (queryName: string, queryId: string) => {
  navigator.clipboard
    .writeText(queryId)
    .then(() => {
      showToastConfirmation({
        body: `${queryName} ID copied successfully!`,
        duration: 2000,
      });
    })
    .catch((error) => {
      console.error("Failed to copy text:", error);
    });
};

/**
 *  Renders a modal to confirm the deletion of a user query.
 * @param modalRef - Reference to the modal component.
 * @param selectedQuery - The currently selected query for deletion.
 * @param handleDelete - Function to handle the deletion workflow.
 * clean up the internal state after deletion.
 * @param queries - The current list of user queries.
 * @param setQueries - Function to update the state of queries.
 * @param context - The data context used to update shared state.
 * @param setSelectedQuery - Function to update the currently selected query to
 * @returns The JSX element for the modal.
 */
export const renderModal = (
  modalRef: RefObject<ModalRef | null>,
  selectedQuery: SelectedQueryDetails | null,
  handleDelete: (
    queryName: string | undefined,
    queryId: string | undefined,
    queries: CustomUserQuery[],
    setQueries: React.Dispatch<
      React.SetStateAction<CustomUserQuery[] | undefined>
    >,
    context: DataContextValue | undefined,
  ) => void,
  queries: CustomUserQuery[],
  setQueries: React.Dispatch<
    React.SetStateAction<CustomUserQuery[] | undefined>
  >,
  context: DataContextValue,
  setSelectedQuery: React.Dispatch<React.SetStateAction<SelectedQueryDetails>>,
): JSX.Element => {
  // here to fix static build errors when we're not in a browser context
  // if (typeof window === "undefined") {
  //   return <></>;
  // }

  return (
    <DeleteModal
      modalRef={modalRef}
      heading="Confirm Deletion"
      description={`Are you sure you want to delete "${
        selectedQuery ? selectedQuery.queryName : ""
      }"? This action cannot be undone.`}
      onDelete={() => {
        if (selectedQuery) {
          handleDelete(
            selectedQuery.queryName,
            selectedQuery.queryId,
            queries,
            setQueries,
            context,
          );
          setSelectedQuery({ queryName: undefined, queryId: undefined });
        }
      }}
    />
  );
};

export type SelectedQueryDetails = {
  queryName?: string;
  queryId?: string;
  pageMode?: string;
};
export type SelectedQueryState = SelectedQueryDetails | null;
