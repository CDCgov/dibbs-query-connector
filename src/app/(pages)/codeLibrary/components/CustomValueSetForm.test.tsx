import { render, screen } from "@testing-library/react";
import { renderWithUser, RootProviderMock } from "@/app/tests/unit/setup";
import {
  cancerValueSets,
  customValueSets,
  gonorrheaValueSets,
  mockDibbsValueSets,
} from "../../queryBuilding/fixtures";
import CustomValueSetForm from "./CustomValueSetForm";
import { getAllUsers, getUserByUsername } from "@/app/backend/user-management";
import { getAllGroupMembers } from "@/app/backend/usergroup-management";
import { mockAdmin } from "../../userManagement/test-utils";
import { emptyValueSet } from "../utils";
import {
  getCustomValueSetById,
  insertCustomValueSet,
} from "@/app/backend/custom-code-service";
import { getAllValueSets } from "@/app/backend/seeding/service";

jest.mock("next-auth/react");

jest.mock(
  "@/app/ui/components/withAuth/WithAuth",
  () =>
    ({ children }: React.PropsWithChildren) => <div>{children}</div>,
);

jest.mock("@/app/backend/usergroup-management", () => ({
  getAllGroupMembers: jest.fn(),
}));

jest.mock("@/app/backend/user-management", () => ({
  getAllUsers: jest.fn(),
  getUserByUsername: jest.fn(),
}));

jest.mock("@/app/backend/seeding/service", () => ({
  getAllValueSets: jest.fn().mockReturnValue({ items: [] }),
}));

jest.mock("@/app/backend/usergroup-management", () => ({
  getAllUserGroups: jest.fn(),
  getAllGroupMembers: jest.fn(),
}));

jest.mock("@/app/backend/custom-code-service", () => ({
  getCustomValueSetById: jest.fn(),
  insertCustomValueSet: jest.fn(),
}));

describe("Create custom valueSet form", () => {
  beforeAll(() => {
    const valueSets = [
      cancerValueSets,
      gonorrheaValueSets,
      customValueSets,
    ].flat();

    (getAllUsers as jest.Mock).mockResolvedValue({ items: [], totalItems: 0 });
    (getAllGroupMembers as jest.Mock).mockResolvedValue({
      items: [],
      totalItems: 0,
    });
    (getUserByUsername as jest.Mock).mockResolvedValue({ items: [mockAdmin] });

    (getAllValueSets as jest.Mock).mockReturnValue({ items: valueSets });
    (insertCustomValueSet as jest.Mock).mockReturnValue({ success: true });

    (getCustomValueSetById as jest.Mock).mockReturnValue({
      items: customValueSets,
    });
  });

  it("renders an empty form to add a new value set", async () => {
    render(
      <RootProviderMock currentPage="/codeLibrary">
        <CustomValueSetForm
          mode={"create"}
          setMode={jest.fn()}
          activeValueSet={emptyValueSet}
        />
      </RootProviderMock>,
    );
    const header = screen.getAllByRole("heading")[0];
    const nameInput = screen.getAllByRole("textbox")[0];
    const dropdowns = screen.getAllByRole("combobox");
    const addCodeInputs = await screen.findAllByTestId("addCode-inputs");

    expect(header).toHaveTextContent("New value set");
    expect(nameInput).not.toHaveValue();
    dropdowns.forEach((dropdown) => {
      expect(dropdown).not.toHaveValue();
    });
    expect(addCodeInputs[0]).not.toHaveValue();
    expect(document.body).toMatchSnapshot();
  });

  it("renders an additional row of code inputs on button click", async () => {
    const { user } = renderWithUser(
      <RootProviderMock currentPage="/codeLibrary">
        <CustomValueSetForm
          mode={"create"}
          setMode={jest.fn()}
          activeValueSet={emptyValueSet}
        />
      </RootProviderMock>,
    );
    const addCodeButton = (await screen.findAllByRole("button")).filter((b) =>
      b.classList.contains("addCodeBtn"),
    );
    expect(await screen.findAllByTestId("addCode-inputs")).toHaveLength(1);
    await user.click(addCodeButton[0]);
    expect(await screen.findAllByTestId("addCode-inputs")).toHaveLength(2);
  });

  it("switches to edit mode after saving a new value set", async () => {
    const setMode = jest.fn();
    const { user } = renderWithUser(
      <RootProviderMock currentPage="/codeLibrary">
        <CustomValueSetForm
          mode={"create"}
          setMode={setMode}
          activeValueSet={emptyValueSet}
        />
      </RootProviderMock>,
    );

    const header = screen.getAllByRole("heading")[0];
    const nameInput = screen.getAllByRole("textbox")[0];
    const dropdowns = screen.getAllByRole("combobox");
    const actionButton = screen.getAllByRole("button")[0];

    expect(header).toHaveTextContent("New value set");
    expect(actionButton).toHaveTextContent("Save value set");

    await user.type(nameInput, "some name");
    await user.selectOptions(dropdowns[0], "labs");
    await user.selectOptions(dropdowns[1], "http://hl7.org/fhir/sid/cvx");
    expect(document.body).toMatchSnapshot();

    await user.click(actionButton);
    expect(setMode).toHaveBeenCalledWith("edit");
  });
});

describe("Edit custom valueSet form", () => {
  beforeAll(() => {
    (getAllUsers as jest.Mock).mockResolvedValue({ items: [], totalItems: 0 });
    (getAllGroupMembers as jest.Mock).mockResolvedValue({
      items: [],
      totalItems: 0,
    });
    (getUserByUsername as jest.Mock).mockResolvedValue({ items: [mockAdmin] });
  });

  it("renders a form to edit an existing value set", async () => {
    render(
      <RootProviderMock currentPage="/codeLibrary">
        <CustomValueSetForm
          mode={"edit"}
          setMode={jest.fn()}
          activeValueSet={mockDibbsValueSets[0]}
        />
      </RootProviderMock>,
    );

    const header = screen.getAllByRole("heading")[0];
    const nameInput = screen.getAllByRole("textbox")[0];
    const dropdowns = screen.getAllByRole("combobox");
    const addCodeRows = await screen.findAllByTestId("addCode-inputs");
    const codeNum =
      addCodeRows[0].firstElementChild?.getElementsByClassName("usa-input")[0];
    const codeName =
      addCodeRows[0].firstElementChild?.nextElementSibling?.getElementsByClassName(
        "usa-input",
      )[0];

    expect(header).toHaveTextContent("Edit value set");
    expect(nameInput).toHaveValue(mockDibbsValueSets[0].valueSetName);
    expect(dropdowns[0]).toHaveValue(mockDibbsValueSets[0].dibbsConceptType);
    expect(dropdowns[1]).toHaveValue(mockDibbsValueSets[0].system);

    expect(codeName).toHaveValue(mockDibbsValueSets[0].concepts[0].display);
    expect(codeNum).toHaveValue(mockDibbsValueSets[0].concepts[0].code);

    expect(document.body).toMatchSnapshot();
  });
});
