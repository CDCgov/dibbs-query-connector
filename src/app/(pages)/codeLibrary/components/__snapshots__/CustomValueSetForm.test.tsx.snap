// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create custom valueSet form renders an empty form to add a new value set 1`] = `
<body>
  <div>
    <div>
      <div
        class="main-container__wide mainContainer customValueSetForm"
      >
        <a
          aria-label="Back arrow indicating ability to navigate back a page if clicked"
          class="back-link unchanged-color-on-visit text-no-underline"
          data-testid="backArrowLink"
          href="#"
        >
          <svg
            aria-label="Arrow point left indicating return to previous step"
            class="usa-icon"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
           
          Return to Codes
        </a>
        <form>
          <div
            class="header"
          >
            <h1
              class="title"
            >
              New value set
            </h1>
            <button
              class="usa-button"
              data-testid="button"
              type="button"
            >
              Save value set
            </button>
          </div>
          <div
            class="formSection vsDescription"
          >
            <div
              class="formSection__title"
            >
              <h2>
                Value set description
              </h2>
            </div>
            <div
              class="formSection__content"
            >
              <div
                class="formSection__input textInput"
              >
                <label
                  for="vsName"
                >
                  Value set name 
                  (Required)
                </label>
                <input
                  class="usa-input"
                  data-testid="textInput"
                  id="vsName"
                  name="vsName"
                  type="text"
                  value=""
                />
              </div>
              <div
                class="formSection__input dropdown"
              >
                <label
                  for="category"
                >
                  Category (Required)
                </label>
                <select
                  class="usa-select filtersSelect"
                  data-testid="Select"
                  id="category"
                  name="Category"
                >
                  <option
                    disabled=""
                    selected=""
                    value=""
                  />
                  <option
                    value="labs"
                  >
                    Labs
                  </option>
                  <option
                    value="conditions"
                  >
                    Conditions
                  </option>
                  <option
                    value="medications"
                  >
                    Medications
                  </option>
                </select>
              </div>
              <div
                class="formSection__input dropdown"
              >
                <label
                  for="code-system"
                >
                  Code system (Required)
                </label>
                <select
                  class="usa-select filtersSelect"
                  data-testid="Select"
                  id="code-system"
                  name="code-system"
                >
                  <option
                    disabled=""
                    selected=""
                    value=""
                  />
                  <option
                    value="http://hl7.org/fhir/sid/icd-10-cm"
                  >
                    ICD-10
                  </option>
                  <option
                    value="http://hl7.org/fhir/sid/cvx"
                  >
                    CVX
                  </option>
                  <option
                    value="http://www.nlm.nih.gov/research/umls/rxnorm"
                  >
                    RXNORM
                  </option>
                  <option
                    value="http://loinc.org"
                  >
                    LOINC
                  </option>
                  <option
                    value="http://cap.org/eCC"
                  >
                    ECC
                  </option>
                  <option
                    value="http://snomed.info/sct"
                  >
                    SNOMED
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            class="codes"
          >
            <div
              class="formSection__title"
            >
              <h2>
                Codes
              </h2>
            </div>
            <div
              class="formSection__content"
            >
              <div
                class="addCodeContainer"
              >
                <div
                  class="addCodeRow formSection__input emptyCodeRow"
                  data-testid="addCode-inputs"
                >
                  <div
                    class="textInput"
                  >
                    <label
                      for="code-id"
                    >
                      Code #
                    </label>
                    <input
                      class="usa-input"
                      data-testid="textInput"
                      id="code-id-0"
                      name="code-id"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="textInput"
                  >
                    <label
                      for="code-name"
                    >
                      Code name
                    </label>
                    <input
                      class="usa-input"
                      data-testid="textInput"
                      id="code-name-0"
                      name="code-name"
                      type="text"
                      value=""
                    />
                  </div>
                  <svg
                    aria-label="Trash icon indicating deletion of code entry"
                    class="usa-icon usa-icon--size-3 margin-bottom-1 deleteIcon"
                    color="#919191"
                    data-testid="delete-custom-code-0"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </div>
              </div>
              <button
                class="addCodeBtn"
              >
                + Add code
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;

exports[`Create custom valueSet form switches to edit mode after saving a new value set 1`] = `
<body>
  <div>
    <div>
      <div
        class="main-container__wide mainContainer customValueSetForm"
      >
        <a
          aria-label="Back arrow indicating ability to navigate back a page if clicked"
          class="back-link unchanged-color-on-visit text-no-underline"
          data-testid="backArrowLink"
          href="#"
        >
          <svg
            aria-label="Arrow point left indicating return to previous step"
            class="usa-icon"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
           
          Return to Codes
        </a>
        <form>
          <div
            class="header"
          >
            <h1
              class="title"
            >
              New value set
            </h1>
            <button
              class="usa-button"
              data-testid="button"
              type="button"
            >
              Save value set
            </button>
          </div>
          <div
            class="formSection vsDescription"
          >
            <div
              class="formSection__title"
            >
              <h2>
                Value set description
              </h2>
            </div>
            <div
              class="formSection__content"
            >
              <div
                class="formSection__input textInput"
              >
                <label
                  for="vsName"
                >
                  Value set name 
                  (Required)
                </label>
                <input
                  class="usa-input"
                  data-testid="textInput"
                  id="vsName"
                  name="vsName"
                  type="text"
                  value="some name"
                />
              </div>
              <div
                class="formSection__input dropdown"
              >
                <label
                  for="category"
                >
                  Category (Required)
                </label>
                <select
                  class="usa-select filtersSelect"
                  data-testid="Select"
                  id="category"
                  name="Category"
                >
                  <option
                    disabled=""
                    selected=""
                    value=""
                  />
                  <option
                    value="labs"
                  >
                    Labs
                  </option>
                  <option
                    value="conditions"
                  >
                    Conditions
                  </option>
                  <option
                    value="medications"
                  >
                    Medications
                  </option>
                </select>
              </div>
              <div
                class="formSection__input dropdown"
              >
                <label
                  for="code-system"
                >
                  Code system (Required)
                </label>
                <select
                  class="usa-select filtersSelect"
                  data-testid="Select"
                  id="code-system"
                  name="code-system"
                >
                  <option
                    disabled=""
                    selected=""
                    value=""
                  />
                  <option
                    value="http://hl7.org/fhir/sid/icd-10-cm"
                  >
                    ICD-10
                  </option>
                  <option
                    value="http://hl7.org/fhir/sid/cvx"
                  >
                    CVX
                  </option>
                  <option
                    value="http://www.nlm.nih.gov/research/umls/rxnorm"
                  >
                    RXNORM
                  </option>
                  <option
                    value="http://loinc.org"
                  >
                    LOINC
                  </option>
                  <option
                    value="http://cap.org/eCC"
                  >
                    ECC
                  </option>
                  <option
                    value="http://snomed.info/sct"
                  >
                    SNOMED
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            class="codes"
          >
            <div
              class="formSection__title"
            >
              <h2>
                Codes
              </h2>
            </div>
            <div
              class="formSection__content"
            >
              <div
                class="addCodeContainer"
              >
                <div
                  class="addCodeRow formSection__input emptyCodeRow"
                  data-testid="addCode-inputs"
                >
                  <div
                    class="textInput"
                  >
                    <label
                      for="code-id"
                    >
                      Code #
                    </label>
                    <input
                      class="usa-input"
                      data-testid="textInput"
                      id="code-id-0"
                      name="code-id"
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    class="textInput"
                  >
                    <label
                      for="code-name"
                    >
                      Code name
                    </label>
                    <input
                      class="usa-input"
                      data-testid="textInput"
                      id="code-name-0"
                      name="code-name"
                      type="text"
                      value=""
                    />
                  </div>
                  <svg
                    aria-label="Trash icon indicating deletion of code entry"
                    class="usa-icon usa-icon--size-3 margin-bottom-1 deleteIcon"
                    color="#919191"
                    data-testid="delete-custom-code-0"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </div>
              </div>
              <button
                class="addCodeBtn"
              >
                + Add code
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;

exports[`Edit custom valueSet form renders a form to edit an existing value set 1`] = `
<body>
  <div>
    <div>
      <div
        class="main-container__wide mainContainer customValueSetForm"
      >
        <a
          aria-label="Back arrow indicating ability to navigate back a page if clicked"
          class="back-link unchanged-color-on-visit text-no-underline"
          data-testid="backArrowLink"
          href="#"
        >
          <svg
            aria-label="Arrow point left indicating return to previous step"
            class="usa-icon"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
            />
          </svg>
           
          Return to Codes
        </a>
        <form>
          <div
            class="header"
          >
            <h1
              class="title"
            >
              Edit value set
            </h1>
            <button
              class="usa-button"
              data-testid="button"
              type="button"
            >
              Save changes
            </button>
          </div>
          <div
            class="formSection vsDescription"
          >
            <div
              class="formSection__title"
            >
              <h2>
                Value set description
              </h2>
            </div>
            <div
              class="formSection__content"
            >
              <div
                class="formSection__input textInput"
              >
                <label
                  for="vsName"
                >
                  Value set name 
                  (Required)
                </label>
                <input
                  class="usa-input"
                  data-testid="textInput"
                  id="vsName"
                  name="vsName"
                  type="text"
                  value="Fruits"
                />
              </div>
              <div
                class="formSection__input dropdown"
              >
                <label
                  for="category"
                >
                  Category (Required)
                </label>
                <select
                  class="usa-select filtersSelect"
                  data-testid="Select"
                  id="category"
                  name="Category"
                >
                  <option
                    disabled=""
                    value=""
                  />
                  <option
                    value="labs"
                  >
                    Labs
                  </option>
                  <option
                    value="conditions"
                  >
                    Conditions
                  </option>
                  <option
                    selected=""
                    value="medications"
                  >
                    Medications
                  </option>
                </select>
              </div>
              <div
                class="formSection__input dropdown"
              >
                <label
                  for="code-system"
                >
                  Code system (Required)
                </label>
                <select
                  class="usa-select filtersSelect"
                  data-testid="Select"
                  id="code-system"
                  name="code-system"
                >
                  <option
                    disabled=""
                    value=""
                  />
                  <option
                    value="http://hl7.org/fhir/sid/icd-10-cm"
                  >
                    ICD-10
                  </option>
                  <option
                    value="http://hl7.org/fhir/sid/cvx"
                  >
                    CVX
                  </option>
                  <option
                    value="http://www.nlm.nih.gov/research/umls/rxnorm"
                  >
                    RXNORM
                  </option>
                  <option
                    value="http://loinc.org"
                  >
                    LOINC
                  </option>
                  <option
                    value="http://cap.org/eCC"
                  >
                    ECC
                  </option>
                  <option
                    selected=""
                    value="http://snomed.info/sct"
                  >
                    SNOMED
                  </option>
                </select>
              </div>
            </div>
          </div>
          <div
            class="codes"
          >
            <div
              class="formSection__title"
            >
              <h2>
                Codes
              </h2>
            </div>
            <div
              class="formSection__content"
            >
              <div
                class="addCodeContainer"
              >
                <div
                  class="addCodeRow formSection__input"
                  data-testid="addCode-inputs"
                >
                  <div
                    class="textInput"
                  >
                    <label
                      for="code-id"
                    >
                      Code #
                    </label>
                    <input
                      class="usa-input"
                      data-testid="textInput"
                      id="code-id-0"
                      name="code-id"
                      type="text"
                      value="1A"
                    />
                  </div>
                  <div
                    class="textInput"
                  >
                    <label
                      for="code-name"
                    >
                      Code name
                    </label>
                    <input
                      class="usa-input"
                      data-testid="textInput"
                      id="code-name-0"
                      name="code-name"
                      type="text"
                      value="Apple"
                    />
                  </div>
                  <svg
                    aria-label="Trash icon indicating deletion of code entry"
                    class="usa-icon usa-icon--size-3 margin-bottom-1 deleteIcon"
                    color="#919191"
                    data-testid="delete-custom-code-0"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </div>
                <div
                  class="addCodeRow formSection__input"
                  data-testid="addCode-inputs"
                >
                  <div
                    class="textInput"
                  >
                    <label
                      for="code-id"
                    >
                      Code #
                    </label>
                    <input
                      class="usa-input"
                      data-testid="textInput"
                      id="code-id-1"
                      name="code-id"
                      type="text"
                      value="1B"
                    />
                  </div>
                  <div
                    class="textInput"
                  >
                    <label
                      for="code-name"
                    >
                      Code name
                    </label>
                    <input
                      class="usa-input"
                      data-testid="textInput"
                      id="code-name-1"
                      name="code-name"
                      type="text"
                      value="Bannana"
                    />
                  </div>
                  <svg
                    aria-label="Trash icon indicating deletion of code entry"
                    class="usa-icon usa-icon--size-3 margin-bottom-1 deleteIcon"
                    color="#919191"
                    data-testid="delete-custom-code-1"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </div>
                <div
                  class="addCodeRow formSection__input"
                  data-testid="addCode-inputs"
                >
                  <div
                    class="textInput"
                  >
                    <label
                      for="code-id"
                    >
                      Code #
                    </label>
                    <input
                      class="usa-input"
                      data-testid="textInput"
                      id="code-id-2"
                      name="code-id"
                      type="text"
                      value="1C"
                    />
                  </div>
                  <div
                    class="textInput"
                  >
                    <label
                      for="code-name"
                    >
                      Code name
                    </label>
                    <input
                      class="usa-input"
                      data-testid="textInput"
                      id="code-name-2"
                      name="code-name"
                      type="text"
                      value="Coconut"
                    />
                  </div>
                  <svg
                    aria-label="Trash icon indicating deletion of code entry"
                    class="usa-icon usa-icon--size-3 margin-bottom-1 deleteIcon"
                    color="#919191"
                    data-testid="delete-custom-code-2"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                    />
                  </svg>
                </div>
              </div>
              <button
                class="addCodeBtn"
              >
                + Add code
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</body>
`;
